pkg/api/imagebasedgroupupgrades/v1alpha1/types.go

```diff
@@ -0,0 +1,97 @@
+package v1alpha1
+
+import (
+	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:generate=true
+
+// ImageBasedUpgradeAction defines the type for the actions field
+// +kubebuilder:validation:Enum=Abort;Prep;Upgrade;Rollback;Finalize
+type ImageBasedUpgradeAction string
+
+// IBUActions defines the string values for valid actions
+var IBUActions = struct {
+	ImageBasedUpgradeAction
+	Prep     ImageBasedUpgradeAction
+	Upgrade  ImageBasedUpgradeAction
+	Rollback ImageBasedUpgradeAction
+	Abort    ImageBasedUpgradeAction
+	Finalize ImageBasedUpgradeAction
+}{
+	Prep:     "Prep",
+	Upgrade:  "Upgrade",
+	Rollback: "Rollback",
+	Finalize: "Finalize",
+	Abort:    "Abort",
+}
+
+// RolloutStrategy defines how to rollout ibu
+type RolloutStrategy struct {
+	//kubebuilder:validation:Minimum=1
+	MaxConcurrency int `json:"maxConcurrency"`
+	//+kubebuilder:default=240
+	Timeout int `json:"timeout,omitempty"`
+}
+
+// ImageBasedGroupUpgradeSpec defines the desired state of ImageBasedGroupUpgrade
+// +kubebuilder:validation:XValidation:message="Invalid list of actions", rule="self.actions==['Prep'] || self.actions==['Prep','Upgrade'] || self.actions==['Prep','Upgrade','Finalize'] || self.actions==['Rollback'] || self.actions==['Rollback', 'Finalize'] || self.actions==['Upgrade'] || self.actions==['Finalize']"
```

Comment:
Looks like missing self.actions==['Upgrade', 'Finalize']?

It seems like we are not blocking invalid transitions. For example, what happens if the old action is Prep and the new action is Rollback? Or is this handled by the controller?

Reply:
updated.

added validation so that you can only append to the list.